org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeCoordinatorHandler.java

```diff
@@ -873,13 +874,21 @@ public void networkStateUpdated(final ZigBeeTransportState state) {
                 break;
             case OFFLINE:
                 Bridge bridge = getThing();
+
                 // do not try to reconnect if there is a firmware update in progress
                 if (bridge.getStatus() == ThingStatus.OFFLINE
                         && bridge.getStatusInfo().getStatusDetail() == ThingStatusDetail.FIRMWARE_UPDATING) {
                     break;
                 }
+                
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-                startReconnectJobIfNotRunning();
+                
+                // try to reconnect (unless the bridge is being removed or uninitialized)
+                if (!Arrays.asList(ThingStatus.UNINITIALIZED, ThingStatus.REMOVING, ThingStatus.REMOVED)
+                        .contains(bridge.getStatus())) {
```

Comment:
What does ```bridge.getStatus()``` return? Is there a bridge defined for the coordinator? I thought that for a coordinator, there is no bridge defined as the coordinator is the bridge for all of the ZigBee devices?

Maybe I'm wrong though? :)

Reply:
That's correct. A couple of lines above (line 876) there is the statement `Bridge bridge = getThing()`. The coordinator is a Thing of type Bridge. ;-)

Reply:
Ok, I missed that line - sorry.

However, that poses a different question.

You code is now

```
bridge = getThing()
updateStatus(ThingStatus.OFFLINE)
bridge.getStatus()
```

Won't this always return OFFLINE?

Reply:
You're right. The implementation needs to be revised.

We found another error which needs to be fixed. Currently the thing status will be set to OFFLINE regardless of its current status. [This documentation](https://www.eclipse.org/smarthome/documentation/concepts/things.html#status-transitions) describes the allowed status transitions. In our case this means that when the thing status is UNINITIALIZED, REMOVING or REMOVED it will not set the thing status to OFFLINE and not restart the reconnect job. I will change the code accordingly.